src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+
+import StorageUtils from '../../../utils/storage';
+import useMappingPostColorHook from '../../FeedScreen/hooks/useMappingColorHook';
+import {Context} from '../../../context';
+import {checkUserBlock, getOtherFeedsInProfile, getOtherProfile} from '../../../service/profile';
+import {setOtherProfileFeed} from '../../../context/actions/otherProfileFeed';
+
+const useOtherProfileScreenHooks = (targetUserProfileId: string, username: string) => {
+  /**
+   * State
+   */
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [isProfileFetching, setIsProfileFetching] = React.useState<boolean>(true);
+  const [isBlocked, setIsBlocked] = React.useState<boolean | undefined>(undefined);
+  const [isBlocking, setIsBlocking] = React.useState<boolean | undefined>(undefined);
+  const [otherProfileData, setOtherProfileData] = React.useState<any>({});
+  const [, setOtherProfileUserId] = React.useState<string>('');
+  const [postOffset, setPostOffset] = React.useState(0);
+
+  /**
+   * Other Profile Context
+   */
+  const [otherProfileFeeds, dispatchOtherProfile] = (React.useContext(Context) as any)
+    ?.otherProfileFeed;
+  const {feeds} = otherProfileFeeds;
+
+  /**
+   * Hook
+   */
+  const {mappingColorFeed} = useMappingPostColorHook();
+
+  const getCache = async () => {
+    const isBlockedFromCache = StorageUtils.blockedStatus.getForKey(targetUserProfileId) === 'true';
+    const isBlockingFromCache =
+      StorageUtils.blockingStatus.getForKey(targetUserProfileId) === 'true';
+
+    try {
+      const otherProfileDataCache =
+        StorageUtils.otherProfileData.getForKey(targetUserProfileId) || '{}';
+      setOtherProfileData(JSON.parse(otherProfileDataCache));
+      if (otherProfileDataCache !== '{}') {
+        setIsProfileFetching(false);
+      }
+    } catch (e) {
+      console.log('error parsing JSON other profile data', e);
+    }
+
+    try {
+      const feedsCache = StorageUtils.otherProfileFeed.getForKey(targetUserProfileId) || '{}';
+
+      setOtherProfileFeed(JSON.parse(feedsCache), dispatchOtherProfile);
+    } catch (e) {
+      console.log('error parsing JSON other profile feed', e);
+    }
+
+    setIsBlocked(isBlockedFromCache);
+    setIsBlocking(isBlockingFromCache);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `getCache` function, you're using `StorageUtils.blockedStatus.getForKey(targetUserProfileId) === 'true'` to determine if the user is blocked or blocking. This can lead to incorrect results if the value is not exactly 'true'. It would be better to parse the result as a boolean.
```diff
- const isBlockedFromCache = StorageUtils.blockedStatus.getForKey(targetUserProfileId) === 'true';
- const isBlockingFromCache = StorageUtils.blockingStatus.getForKey(targetUserProfileId) === 'true';
+ const isBlockedFromCache = JSON.parse(StorageUtils.blockedStatus.getForKey(targetUserProfileId));
+ const isBlockingFromCache = JSON.parse(StorageUtils.blockingStatus.getForKey(targetUserProfileId));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
Should be no problem, since the value is true or false